File: fsharp/p02.fs
[<EntryPoint>]
let main(argv) = 
  let limit = if Array.length argv > 0 then argv.[0] |> int
              else 4000000
  let mutable sum = 0
  let mutable a = 1
  let mutable b = 1
  let mutable c = 2
  while c < limit do
    sum <- sum + c
    a <- b + c
    b <- c + a
    c <- a + b
  printfn "%i" sum
  0

File: csharp/p02.cs
class P2{
  static void Main(string[] args){
    int limit = 4000000;
    int sum = 0, a = 1, b = 1, c = 2;
    if (args.Length >= 1){
      limit = int.Parse(args[0]);
    }
    while (c < limit){
      sum += c;
      a = b + c;
      b = c + a;
      c = a + b;
    }
    System.Console.Write("{0}\n", sum);
  }
}
